src/App.tsx

```diff
@@ -17,18 +85,21 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onStatusChange={setStatus}
+                onSearch={setSearch}
+                onClear={() => setSearch('')}
+              />
             </div>
 
-            <div className="block">
-              <Loader />
-              <TodoList />
-            </div>
+            <div className="block">{getTodosListView()}</div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal todo={selectedTodo} onClose={() => setSelectedTodo(null)} />
```

Comment:
The `TodoModal` component is being passed a prop named `todo`, which is correct according to the task requirements. However, ensure that this aligns with the `TodoModal` component's expected prop name. If the component expects `todos`, it should be singular (`todo`) to reflect that it represents a single todo item.